src/library/Query.js

```diff
@@ -33,11 +45,49 @@ function query(httpMethod, deployment, apiKey, qry, payload) {
 }
 
 function limitQuery(httpMethod, deployment, apiKey, queryPath, limit, offset, payload, address) {
-  // validate and normalize limit and offset
-  const queryOptions = buildQueryOptions(limit, offset, address);
+  const separationLimit = 30;
+  const limitChecked = checkLimit(limit);
+  if (limitChecked > separationLimit) {
+    const results = {
+      status: null,
+      message: null,
+    };
+    // MBQUERY, MBCUSTOMQUERY has rows in result but MBEVENTS has not
+    const hasRows = !/^events$/.test(queryPath);
+    const rows = [];
 
-  // query
-  const results = query(httpMethod, deployment, apiKey, queryPath + queryOptions, payload);
+    let offsetNext = offset;
+    while (true) {
+      // validate and normalize limit and offset
+      const queryOptions = buildQueryOptions(separationLimit, offsetNext, address);
 
-  return results;
+      // query
+      const queryResult = query(httpMethod, deployment, apiKey, queryPath + queryOptions, payload);
+      results.status = queryResult.status;
+      results.message = queryResult.message;
+
+      const queryRows = hasRows ? queryResult.result.rows : queryResult.result;
+      const preLength = rows.length + queryRows.length;
+      const currentLength = queryRows.length;
+      if (currentLength < 1) {
+        break;
+      } else if (preLength > limitChecked) {
+        // Get rid of the overflowed if actual data is more than limitChecked(limit)
+        const cutOff = currentLength - (preLength - limitChecked);
+        rows.push(...queryRows.slice(0, cutOff));
+        break;
+      } else {
+        rows.push(...queryRows);
+      }
+
+      offsetNext += separationLimit;
+    }
+
+    results.result = hasRows ? { rows } : rows;
+
+    return results;
+  }
+
+  const queryOptions = buildQueryOptions(limitChecked, offset, address);
+  return query(httpMethod, deployment, apiKey, queryPath + queryOptions, payload);
```

Comment:
Perhaps remove this special case and simplify the loop slightly?

Something like:

```js
const clampedLimit = min(separationLimit, limit);

while (rows.length < limit) {
   // ...
}

// Get rid of the overflowed ...

return results
```